plugins/extraction/extraction.ml

```diff
@@ -488,7 +488,7 @@ and extract_really_ind env kn mib =
         if p.ip_logical then raise (I Standard);
         if not (Int.equal (Array.length p.ip_types) 1) then raise (I Standard);
         let typ = p.ip_types.(0) in
-        let l = List.filter (fun t -> not (isTdummy (expand env t))) typ in
+        let l = List.filter (fun t -> conservative_types () || not (isTdummy (expand env t))) typ in
```

Comment:
To clearly show the intent, I think you should extrude the check outside of the filter as in
```
let l = if conservative_types () then l else ...
```

Reply:
You are right, that's better. Done.